File: tests-python/pages/header_component.py
from __future__ import annotations

import allure
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC

from pages.base import BasePage


class HeaderComponent(BasePage):
    ROOT = (By.CSS_SELECTOR, "[data-testid='header']")

    @allure.step("Verify header is mounted")
    def should_be_mounted(self) -> HeaderComponent:
        self.wait_visible(*self.ROOT)
        return self

    @allure.step("Verify header nav '{active_testid}' is the only active item")
    def should_have_active_nav(self, active_testid: str) -> HeaderComponent:
        self.should_be_mounted()
        active = self.wait().until(
            EC.visibility_of_element_located((By.CSS_SELECTOR, f"[data-testid='{active_testid}']"))
        )

        def _is_active(_driver) -> bool:
            cls = active.get_attribute("class") or ""
            aria = active.get_attribute("aria-current")
            return "is-active" in cls.split() and aria == "page"

        self.wait().until(_is_active)
        current = self.driver.find_elements(
            By.CSS_SELECTOR, "[data-testid='header-nav'] a[aria-current='page']"
        )
        assert len(current) == 1
        return self

File: java/gradle-junit5-selenium/src/test/java/pages/HeaderComponent.java
package pages;

import config.TestConfig;
import io.qameta.allure.Step;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class HeaderComponent extends BasePage {

    private static final By ROOT = By.cssSelector("[data-testid='header']");
    private static final By ACTIVE_NAV = By.cssSelector("[data-testid='header-nav'] a[aria-current='page']");

    public HeaderComponent(WebDriver driver, TestConfig config) {
        super(driver, config);
    }

    @Step("Verify header is mounted")
    public HeaderComponent shouldBeMounted() {
        waitVisible(ROOT);
        return this;
    }

    @Step("Verify header nav '{activeTestid}' is the only active item")
    public HeaderComponent shouldHaveActiveNav(String activeTestid) {
        shouldBeMounted();
        By activeLocator = By.cssSelector("[data-testid='" + activeTestid + "']");
        WebElement active = waitVisible(activeLocator);
        driverWait().until(d -> {
            String cls = active.getAttribute("class");
            String aria = active.getAttribute("aria-current");
            boolean hasClass = cls != null && java.util.Arrays.asList(cls.split("\\s+")).contains("is-active");
            return hasClass && "page".equals(aria);
        });
        driverWait().until(d -> d.findElements(ACTIVE_NAV).size() == 1);
        return this;
    }
}
